torchgeo/transforms/indices.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""TorchGeo indices transforms."""
+
+from typing import Dict
+
+import torch
+from torch import Tensor
+from torch.nn import Module  # type: ignore[attr-defined]
+
+# https://github.com/pytorch/pytorch/issues/60979
+# https://github.com/pytorch/pytorch/pull/61045
+Module.__module__ = "torch.nn"
+
+
+def ndvi(red: Tensor, nir: Tensor) -> Tensor:
+    """Compute Normalized Different Vegetation Index (NDVI).
+
+    Args:
+        red: tensor containing red band
+        nir: tensor containing nir band
+
+    Returns:
+        tensor containing computed NDVI values
+    """
+    return (nir - red) / (nir + red)
+
+
+class NDVI(Module):  # type: ignore[misc,name-defined]
```

Comment:
According to the [source code](https://github.com/kornia/kornia/blob/master/kornia/augmentation/container/augment.py#L87), our transforms need to subclass `_AugmentationBase` in order to be used with `AugmentationSequential`.

Reply:
True, however computing an index and appending it as an additional channel/band is probably not considered an `augmentation` according to the `_AugmentationBase` class which requires probability of occurrence parameters like `p`, and `p_batch`. Although we could force these values to be `1.0` by default, I think that it's possible that `AugmentationSequential` is flexible enough to simply take in `nn.Module` transforms as well.

Reply:
> I think that it's possible that AugmentationSequential is flexible enough to simply take in nn.Module transforms as well.

It might run, but mypy won't allow it.